exercises/flatten-array/src/example.c

```diff
@@ -0,0 +1,48 @@
+#include "flatten_array.h"
+#include <stdlib.h>
+#include <stdio.h>
+
+enum modes { READ, WRITE };
+
+static size_t flat_array_pos;
+static int *flat_array = NULL;
+
+static void parse_array(item_t * head, enum modes mode)
+{
+   if (head->type == ELEMENT) {
+      if (mode == WRITE) {
+         flat_array[flat_array_pos] = head->value;
+      }
+      flat_array_pos++;
+   }
+
+   if (head->array_elements != NULL)
+      parse_array(head->array_elements, mode);
+
+   if (head->next != NULL)
+      parse_array(head->next, mode);
+}
+
+int *flatten(item_t * head)
+{
+   if (head == NULL)
+      return NULL;
+
+   flat_array_pos = 0;
+   parse_array(head, READ);
+   const size_t flat_array_len = flat_array_pos;
+
+   if (flat_array_len == 0)
+      return 0;
+
+   flat_array = malloc(sizeof(int) * flat_array_len);
```

Comment:
When using the `sizeof` operator it is generally better to apply it to a variable rather than directly to the type (e.g. `malloc(sizeof(*flat_array) * flat_aray_len);`) as this decouples the _implementation_ of `flatten()` from the types (i.e. there will be less places that code would need changing if the type of `flat_array` needed changing).